fix: LEAP-1815: Fix initial image tool selection logic

Gondragos opened the pull request:
### PR fulfills these requirements
- [x] Commit message(s) and PR title follow the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] The best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)


#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend



### Describe the reason for change
There was a conflict between two features: `default tool` and `preserved tool`. Due to logic inconsistency, the second one annulated the first one even where there is no resolution about the selected tool yet.

`isInitial` flag is added to prevent a situation when the default tool sets a new preserved tool state to local storage before it was handled at all.


#### What does this fix?
It allows the default tool to be selected in case if the preserved tool was not found in the config.


### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)



### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [x] integration
- [ ] unit



### Which logical domain(s) does this change affect?
`Image Tools`

Gondragos committed:
fix: LEAP-1815: Correct initial image tool selection logic

Adjusted the return statement placement within the tool selection logic to ensure proper flow and avoid unintended behavior.

Gondragos committed:
test: Add integration tests for initial tool selection in image segmentation

Introduces tests covering scenarios for default tool selection, preserved tool state, and invalid tool cases during initialization.

bmartel commented on web/libs/editor/src/tools/Manager.js:
```diff
@@ -74,8 +74,8 @@ class ToolsManager {
       if (tool.fullName === this.preservedTool && tool.setSelected) {
         this.unselectAll();
         this.selectTool(tool, true);
+        return;
```
Wow 👏

MihajloHoma commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/13174465250)
> Successfully merged:  create mode 100644 docs/source/guide/prompts_keys.md

MihajloHoma committed:
Merge branch 'develop' into 'fb-LEAP-1815/initial-image-tool'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13174465250

Gondragos committed:
fix: Prevent losing preserved state on initialization

Add an `isInitial` flag to differentiate between initial and user-triggered tool selection. This prevents unnecessary actions during default tool assignment or preserved tool restoration. Updates ensure proper handling of state transitions with minimal side effects.
